### PR TITLE
Bug 1755586: Removing simpleMacvlanConfig from CNO not delete old network-attachment-definition

### PR DESCRIPTION
**Pull request by s1061123**
This change adds to remove net-attach-def object if corresponding additionalNetwork is removed in CNO CRD object.

### CODE REVIEW

File: pkg/network/render.go [hunk 1]
```diff
@@ -389,3 +393,48 @@ func RenderMultusAdmissionController(conf *operv1.NetworkSpec, manifestDir strin
 	out = append(out, objs...)
 	return out, nil
 }
+
+func ClearNetAttachDef(client cli.Client, prev *operv1.NetworkSpec, conf *operv1.NetworkSpec) error {
```
Comment (squeed) [inline pkg/network/render.go]: nit: This should be in `pkg/controller/operconfig_controller`, not `pkg/network`.

Comment (squeed) [reply pkg/network/render.go]: also, can you add a docblock describing this?

File: pkg/network/render.go [hunk 2]
```diff
@@ -389,3 +393,48 @@ func RenderMultusAdmissionController(conf *operv1.NetworkSpec, manifestDir strin
...
+
+	for _, netAttach := range deleteNetAttachDefs {
+		log.Printf("delete net-attach-def/%s namespace: %s", netAttach.Name, netAttach.Namespace)
+
+		if netAttach.Namespace == "" {
+			netAttach.Namespace = "default"
+		}
+
+		obj := &netattachv1.NetworkAttachmentDefinition{}
+		err := client.Get(context.TODO(), cli.ObjectKey{
```
Comment (squeed) [inline pkg/network/render.go]: Why bother getting it? I don't think you need to.

File: pkg/network/render.go [hunk 3]
```diff
@@ -389,3 +393,48 @@ func RenderMultusAdmissionController(conf *operv1.NetworkSpec, manifestDir strin
...
+			Namespace: netAttach.Namespace,
+			Name:      netAttach.Name,
+		}, obj)
+		if err != nil {
+			log.Printf("failed to get net-attach-def %s in namespace %s: %v", netAttach.Name, netAttach.Namespace, err)
+			break
+		}
+
```
Comment (squeed) [inline pkg/network/render.go]: You should only delete the NetAttachDef if the OwnerReference is set to ours. That way we won't accidentally step on anyone who re-creates it.

File: pkg/network/render.go [hunk 4]
```diff
@@ -389,3 +393,48 @@ func RenderMultusAdmissionController(conf *operv1.NetworkSpec, manifestDir strin
...
+		if netAttach.Namespace == "" {
+			netAttach.Namespace = "default"
+		}
+
+		obj := &netattachv1.NetworkAttachmentDefinition{}
+		err := client.Get(context.TODO(), cli.ObjectKey{
+			Namespace: netAttach.Namespace,
+			Name:      netAttach.Name,
+		}, obj)
+		if err != nil {
```
Comment (squeed) [inline pkg/network/render.go]: Don't print an error if `apierrors.IsNotFound()`

File: pkg/network/render.go [hunk 5]
```diff
@@ -389,3 +393,48 @@ func RenderMultusAdmissionController(conf *operv1.NetworkSpec, manifestDir strin
...
+			log.Printf("failed to get net-attach-def %s in namespace %s: %v", netAttach.Name, netAttach.Namespace, err)
+			break
+		}
+
+		err = client.Delete(context.TODO(), obj)
+		if err != nil {
+			log.Printf("failed to delete net-attach-def %s in namespace %s: %v", netAttach.Name, netAttach.Namespace, err)
```
Comment (squeed) [inline pkg/network/render.go]: Don't print an error if apierrors.IsNotFound()

**Comment by squeed** [comment]
overall I think this is OK. We have a more robust mechanism for deleting stale objects in #306, but that won't be backported to previous versions, so I would be OK with this also being merged.

File: pkg/network/render.go (hunk 1 above)
Comment (s1061123) [reply pkg/network/render.go]: changed.

File: pkg/network/render.go (hunk 4 above)
Comment (s1061123) [reply pkg/network/render.go]: fixed.

File: pkg/network/render.go (hunk 5 above)
Comment (s1061123) [reply pkg/network/render.go]: Done.

**Comment by s1061123** [comment]
@dcbw PTAL?

**Comment by s1061123** [comment]
@dcbw PTAL?

**Comment by s1061123** [comment]
@dcbw PTAL?

**Pull request closed without merging**